### PR TITLE
`isinstance` checks should not use generics from typing

### PR DESCRIPTION
**Pull request by HaydenCognite**
## Description
See title, stolen from [doge34](https://cognitedata.atlassian.net/browse/DOGE-34). @haakonvt to check if I missed anything or misunderstood something. I also made a few other changes that may or may not be welcome. :)

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/_api/raw.py [hunk 1]
```diff
@@ -495,12 +495,9 @@ def retrieve(self, db_name: str, table_name: str, key: str) -> Row | None:
...
         if columns is None:
             return None
-        if not isinstance(columns, List):
+        if not isinstance(columns, list):
             raise ValueError("Expected a list for argument columns")
-        if len(columns) == 0:
-            return ","
-        else:
-            return ",".join([str(x) for x in columns])
+        return "," if len(columns) == 0 else ",".join([str(x) for x in columns])
```
Comment (HaydenCognite) [inline cognite/client/_api/raw.py:500]: I know not everyone is a fan of this so let me know if it should be reverted.

Comment (haakonvt) [inline cognite/client/_api/raw.py:500]: Lgtm, I would switch it around though, also don't need to build a temporary list:
```suggestion
        return ",".join(str(x) for x in columns) if columns else ","
```

**Review by erlendvollset — APPROVED** [review]
LGTM

**Review by haakonvt — APPROVED** [review]
Lgtm, just had one comment

File: cognite/client/_api/raw.py (hunk 1 above)
Comment (haakonvt) [reply cognite/client/_api/raw.py:500]: As a side note, I generally dislike conditional expressions (except when they are super-short) 😉

Comment (HaydenCognite) [reply cognite/client/_api/raw.py:500]: OK, I will take a mix of your answer and @haakonvt s answer by reverting it, but also removing the list creation. Hope that is fine with both of ye.

Comment (HaydenCognite) [reply cognite/client/_api/raw.py]: No worries, I will take a mix of both of your comments (removing the list instantiation and returning to the if-else branch).

**Pull request merged**